File: typescript/pages/AdminPage.ts
import { Page, Locator, expect } from '@playwright/test';

export class AdminPage {
  private readonly roleDropdown: Locator;
  private readonly searchButton: Locator;
  private readonly resultsContainer: Locator;

  constructor(private page: Page) {
    this.roleDropdown     = page.locator('.oxd-select-text--after').first();
    this.searchButton     = page.getByRole('button', { name: 'Search' });
    this.resultsContainer = page.locator('#app');
  }

  async goto() {
    await this.page.getByRole('link', { name: 'Admin' }).click();
  }

  async filterByRole(role: string) {
    await this.roleDropdown.click();
    await this.page.getByRole('option', { name: role }).click();
  }

  async search() {
    await this.searchButton.click();
  }

  async verifyResultsFound() {
    await expect(this.resultsContainer).toContainText('Record Found');
  }
}

File: python/pages/admin_page.py
from playwright.sync_api import Page, Locator, expect


class AdminPage:
    def __init__(self, page: Page):
        self._page = page
        self._role_dropdown:     Locator = page.locator('.oxd-select-text--after').first
        self._search_button:     Locator = page.get_by_role('button', name='Search')
        self._results_container: Locator = page.locator('#app')

    def goto(self):
        self._page.get_by_role('link', name='Admin').click()

    def filter_by_role(self, role: str):
        self._role_dropdown.click()
        self._page.get_by_role('option', name=role).click()

    def search(self):
        self._search_button.click()

    def verify_results_found(self):
        expect(self._results_container).to_contain_text('Record Found')
